care/utils/assetintegration/onvif.py

```diff
@@ -1,13 +1,149 @@
 import enum
+import json
 
-from rest_framework.exceptions import ValidationError
+from django.core.cache import cache
+from rest_framework.exceptions import PermissionDenied, ValidationError
 
+from care.users.models import User
 from care.utils.assetintegration.base import BaseAssetIntegration
 
 
 class OnvifAsset(BaseAssetIntegration):
     _name = "onvif"
 
+    class UsageManager:
```

Comment:
we should move this out of the onvifasset class, later on this can be abstracted for other assets if needed